airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingConsumerFactory.java

```diff
@@ -149,13 +151,13 @@ private static RecordWriter recordWriterFunction(final JdbcDatabase database,
       final WriteConfig writeConfig = pairToWriteConfig.get(pair);
       final String schemaName = writeConfig.getOutputSchemaName();
       final String tableName = writeConfig.getOutputTableName();
-      final String stageName = namingResolver.getStageName(schemaName, tableName);
+      final String path = namingResolver.getStagingPath(schemaName, tableName, CURRENT_SYNC_PATH);
```

Comment:
When there are multiple "flush" on the same stream during a sync, are all batches/files written to the same path / same object in the stage area?

Reply:
The path is like a virtual folder. It is similar to S3:
https://docs.snowflake.com/en/user-guide/data-load-considerations-stage.html

We always use the temp filename as the file under the folder:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeStagingSqlOperations.java#L45